### PR TITLE
Adds acct foreign-constraint to kyc & address changeset

### LINKED ISSUE #82: Constraint error when using "upsertKyc" mutation
Error:
> constraint error when attempting to insert struct:    * kyc_data_account_id_fkey (foreign_key_constraint)If you would like to stop this constraint violation from raising anexception and instead add it as an error to your changeset, pleasecall `foreign_key_constraint/3` on your changeset with the constraint`:name` as an option.The changeset has not defined any constraint.

Query:
> {"query":"mutation {\n  upsertKyc2069213328: upsertKyc(input: {accountId: \"rodrigoky14\", countryId: \"1\", document: \"1231231231\", documentType: \"nite\", phone: \"11948340013\", userType: \"natural\"}) {\n    __typename0: __typename\n  }\n}"}

### PR DESCRIPTION
**Pull request by victorgdev**
## What issue does this PR close
Closes #82 

## Changes Proposed ( a list of new changes introduced by this PR)
- Adds the account_id foreign constraint to the KYC and Address changeset

## How to test ( a list of instructions on how to test this PR)
- Go to the GraphiQL Playground 
- Run the upsertKYC and/or upsertAddress mutations

**Examples For Testing**

```
// UpsertKYC

mutation {
    upsertKyc( input: 
      { accountId: "andreymisk",  // account does not exist
      countryId: "1", 
      document: "1231231231", 
      documentType: "nite", 
      phone: "11948340013", 
      userType: "natural"})
      { phone
      country { name } }
}

// UpsertAddress

mutation {
   upsertAddress( input: 
      { accountId: "andreymisk",
      cityId: "1",
      countryId: "1", 
      neighborhoodId: "1",
      number: "123",
      stateId: "1",
      street: "Elm street",
      zip: "10101"})
      { country { name } }
}
```

## Mutation Results
BEFORE
image

AFTER
image

### CODE REVIEW

**Comment by victorgdev** [comment]
@lucca65 , @RodrigoLeiteF, @amiskov  can I get some feedback on the solution, only fixed the constraint error and not the error messaging for the frontend handling, which is related to [this issue](https://github.com/cambiatus/backend/issues/84)

File: lib/cambiatus/kyc/kyc_data.ex [hunk 1]
```diff
@@ -34,8 +34,12 @@ defmodule Cambiatus.Kyc.KycData do
     |> validate_document_type()
     |> validate_format(:phone, ~r/[1-9]{1}\d{3}-?\d{4}/)
     |> validate_document()
+    |> foreign_key_constraint(:account_id)
+    |> foreign_key_constraint(:country_id)
   end
 
+
+
```
Comment (amiskov) [inline lib/cambiatus/kyc/kyc_data.ex:42]: There are 2 extra lines, they're unnecessary.

File: lib/cambiatus/kyc/kyc_data.ex [hunk 2]
```diff
@@ -34,8 +34,12 @@ defmodule Cambiatus.Kyc.KycData do
     |> validate_document_type()
     |> validate_format(:phone, ~r/[1-9]{1}\d{3}-?\d{4}/)
     |> validate_document()
+    |> foreign_key_constraint(:account_id)
+    |> foreign_key_constraint(:country_id)
```
Comment (amiskov) [inline lib/cambiatus/kyc/kyc_data.ex:38]: My suggestion here is to move these `foreign_key_constraint` up and place them right after `cast(...)` like it's done in `address.ex`. Functions in this pipe are run one-after-another and if there are any constrain violation then we don't need any validation at all.

**Review by amiskov — CHANGES_REQUESTED** [review]
All looks good for me, there are just a couple of suggestions.

File: lib/cambiatus/kyc/kyc_data.ex (hunk 1 above)
Comment (amiskov) [reply lib/cambiatus/kyc/kyc_data.ex:42]: BTW, you can run `mix format lib/cambiatus/kyc/kyc_data.ex` to get rid of any formatting issues. In VSCode there's also an option to format files automatically on save:

image

**Comment by victorgdev** [comment]
Please refer to [PR#88](https://github.com/cambiatus/backend/pull/88)

File: .gitignore [hunk 3]
```diff
@@ -1,26 +0,0 @@
-# App artifacts
```
Comment (RodrigoLeiteF) [inline .gitignore:1]: We need this file

File: config/dev.exs [hunk 4]
```diff
@@ -1,58 +0,0 @@
-use Mix.Config
```
Comment (RodrigoLeiteF) [inline config/dev.exs:1]: This one too

File: config/test.exs [hunk 5]
```diff
@@ -1,44 +0,0 @@
-use Mix.Config
```
Comment (RodrigoLeiteF) [inline config/test.exs:1]: And this one

**Pull request closed without merging**